### ISSUE TITLE
Add Almond support

### ISSUE THREAD

**Issue by ianb** [mozilla/firefox-voice#835]
[Almond](https://almond.stanford.edu/) [has a web interface](https://almond.stanford.edu/me) (signup required). 

We could use it a bit like we use Google, though when or where we would send queries there instead of Google, I'm not sure. We could have a configurable fallback?

Example interaction:

![image](https://user-images.githubusercontent.com/44390/72090951-afac9480-32d4-11ea-8989-286c65d9de98.png)

**Comment by maitrella** [comment]
I was unaware of this project. Thanks for showing us such interesting things!.

**Comment by 12people** [comment]
Almond is completely open-source: https://github.com/stanford-oval/almond-cloud
It could just be integrated into Firefox Voice, no account or website redirection needed.

Alternatively, Firefox Voice could just plug into [Almond's Thingpedia](https://almond.stanford.edu/thingpedia), which is an open-source repository of commands.

**Comment by davidak** [comment]
I think the end goal should be to have a voice controlled personal assistant integrated into the operating system. Running as a service. Or even as an individual device. That can then interact with other software like Firefox. When we have that, firefox-voice is not needed anymore.

So i think this project should focus on firefox specific actions and in best case, host them on Thingpedia (or something similar, haven't looked into it), so other services can also control firefox. We should not duplicate efforts!

Almond and Mycroft might provide a useful desktop personal assistant soon.

**Comment by 12people** [comment]
> I think the end goal should be to have a voice controlled personal assistant integrated into the operating system. Running as a service. Or even as an individual device. That can then interact with other software like Firefox. When we have that, firefox-voice is not needed anymore.

I agree only partly. The goal, IMHO, should be to have a native **privacy-minded** voice assistant that **doesn't rely on proprietary technology**.

I personally would never consider using a proprietary tool like Amazon's Alexa or Google Assistant where I wouldn't have control over my privacy.

> So i think this project should focus on firefox specific actions and in best case, host them on Thingpedia (or something similar, haven't looked into it), so other services can also control firefox. We should not duplicate efforts!

Agreed on not duplicating efforts unnecessarily.

However, the only reason that Firefox Voice exists is because the assistant defined above doesn't exist on most platforms.
MyCroft is still Linux-only after more than 4 years of development. Almond is only working on command interpretation AFAIK and eschews speech-to-text completely.

Firefox Voice is a voice assistant, and not just a set of APIs. Therefore, I think it'd be very useful if Almond's stellar command interpretation was made to work with Firefox's speech-to-text.

In fact, if we're talking about not duplicating efforts, the ideal would be to have Firefox Voice become an STT interface for Almond and add Firefox-specific skills to Almond's Thingpedia.

**Comment by davidak** [comment]
>I agree only partly. The goal, IMHO, should be to have a native **privacy-minded** voice assistant **that doesn't rely on proprietary technology**.

sure. I take that as a basic requirement in this thread :smile: 

>However, the only reason that Firefox Voice exists is because the assistant defined above doesn't exist on most platforms.

yes, i agree

>if we're talking about not duplicating efforts, the ideal would be to have Firefox Voice become an STT interface 

i think that should also be a system service like TTS for many years on macOS, Windows and Linux (through Speech Dispatcher). i think it will not take very long until we have that with Mozillas efforts, at least on linux where i can package and configure it myself
